brainscore_language/models/earley_parser/parser.py

```diff
@@ -0,0 +1,327 @@
+import logging
+from pathlib import Path
+from typing import Tuple, Union, List, Dict, Optional
+
+import numpy as np
+import xarray as xr
+import nltk
+from nltk.grammar import PCFG, Nonterminal, is_terminal
+from nltk.corpus.reader import BracketParseCorpusReader
+from nltk.parse.pchart import Chart
+
+from brainio.assemblies import DataAssembly, BehavioralAssembly
+from brainscore_language.artificial_subject import ArtificialSubject
+from brainscore_language.utils import fullname
+
+from brainscore_language.models.earley_parser.utils import (
+    ProbabilisticEarleyChartParser,
+)
+
+PARSING_TRACE = 0  # how verbose the tracing output should be while parsing a text
+
+
+class EarleyParserSubject(ArtificialSubject):
+    def __init__(
+        self,
+        model_id: str,
+    ):
+        """
+        :param model_id: the model id i.e. name
+        """
+        self._logger = logging.getLogger(fullname(self))
+        self.model_id = model_id
```

Comment:
since the `model_id` doesn't affect anything else in constructing the object, I would hardcode:
```suggestion
    ):
        self._logger = logging.getLogger(fullname(self))
        self.model_id = 'earley-parser'
```

Unless you want the grammar to be dynamic too, then I suppose the `treebank_path` should be an argument along with the `model_id`